internal/search/query/searchquery_test.go

```diff
@@ -131,6 +132,38 @@ func TestQuery_StringValues(t *testing.T) {
 	})
 }
 
+func TestQuery_Validate(t *testing.T) {
+	cases := []struct {
+		Name       string
+		Query      string
+		SearchType SearchType
+		Want       string
+	}{
+		{
+			Name:       `Structural search validates`,
+			Query:      `patterntype:structural ":[_]"`,
+			SearchType: SearchTypeStructural,
+			Want:       `the parameter "case:" is not valid for structural search`,
+		},
+		{
+			Name:       `Structural search incompatible with "case:"`,
+			Query:      `patterntype:structural case:yes ":[_]"`,
+			SearchType: SearchTypeStructural,
+			Want:       `the parameter "case:" is not valid for structural search, matching is always case-sensitive`,
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.Name, func(t *testing.T) {
+			q, _ := ParseAndCheck(tt.Query)
+			if got := Validate(q, tt.SearchType); got != nil {
+				if diff := cmp.Diff(got.Error(), tt.Want); diff != "" {
+					t.Fatal(diff)
```

Comment:
No need to fatal, since each test case is independent.

```suggestion
					t.Error(diff)
```

Reply:
I often find `Fatal` is more convenient when a test actually fails. It stops further process as the whole testing is going fail anyway, thus less nosies for me.

Reply:
You can just use the `-failfast` flag for that.

```
 -failfast
        do not start new tests after the first test failure
```

Reply:
TIL! But I still feel little benefit using `t.Error`.

Reply:
The main benefit is that if you have multiple tests that will fail in your table, you won't see logs from any but the first till you've "fixed" the first one. Often it's helpful for diagnosing a root cause (which is often shared) to see the logs from _all_ the failing tests before exiting the test. YMMV, though, as you say.